### ISSUE TITLE
Pressing any accent before Escape makes Escape fire two "pressed" events

### ISSUE THREAD

**Issue by PetePete1984** [godotengine/godot#16858]
**Godot version:**
tested in 3.0 stable and 2.1.4 stable

**OS/device including version:**
Windows 10 64bit (desktop build), including Fall Creators' Update

**Issue description:**
While listening for input, when you press the Circumflex accent before pressing Escape, you'll get two key press events for the Escape key, both as InputEvents and as Actions.
Pressing other keys like Space or A after Circumflex does not exhibit that behaviour.

This is a mild inconvenience on German keyboards and possibly others, where the Circumflex key is located below Escape (sometimes directly, like on this here Laptop).

Also happens with other accent keys like accent acute or accent grave (the tick / backtick characters), although these are in a different spot so it's less easy to fat-finger them.

Debug print from Godot 2:

> Circumflex + ESC
> Device 0 ID 722 Event: Key Unicode:  Scan: 92 Echo: False PressedTrue Mod:
> Device 0 ID 723 Event: Key Unicode:  Scan: 92 Echo: False PressedFalse Mod:
> Device 0 ID 726 Event: Key Unicode: ^ Scan: 16777217 Echo: False PressedTrue Mod:
> Device 0 ID 727 Event: Key Unicode:  Scan: 16777217 Echo: False PressedTrue Mod:
> Device 0 ID 728 Event: Key Unicode:  Scan: 16777217 Echo: False PressedFalse Mod:
> 
> Circumflex + A:
> Device 0 ID 774 Event: Key Unicode:  Scan: 92 Echo: False PressedTrue Mod:
> Device 0 ID 775 Event: Key Unicode:  Scan: 92 Echo: False PressedFalse Mod:
> Device 0 ID 777 Event: Key Unicode: â Scan: 65 Echo: False PressedTrue Mod:
> Device 0 ID 778 Event: Key Unicode:  Scan: 65 Echo: False PressedFalse Mod:

The attached minimal project shows the behaviour in Godot 3 using var2str on the event; circumflex produces 2 events, the Escape key again produces 3.

The workaround on my end will be to discard Escape key events with the Circumflex char in their unicode property, will have to see if using actions still remains functional or if I'll have to rely on InputEventKey.

**Steps to reproduce:**
- any Node must have an _input(event) function
- with the default keybindings, Escape is bound to the action "ui_cancel"
- have _input react to either event.is_action_pressed("ui_cancel") or check for InputEventKey and the correct Scancode, then for example print anything to the console
- press circumflex, then escape
- the print will happen twice; if you print the key events, you'll get 2 for the accent and 3 for escape

**Minimal reproduction project:**
[Circumflex.zip](https://github.com/godotengine/godot/files/1740551/Circumflex.zip)

**Comment by Duehok** [comment]
This issue is actually a bit worse: the same problem appears with tab, enter and backspace. Tab and backspace are next to problematic keys.
The problem also appears when using one of the accent keys twice: for example ^then ´, or twice ^. If anybody ever wants to use more than the 1-0 keys as shortcuts and instead use the whole top row there will be problems for german users.

Tested on win 8.1 godot 3.0 stable with the reproduction project.

**Comment by vnen** [comment]
This is a thing with the dead keys. When you press the circumflex (a dead key) it keeps hanging until another key is pressed, so it can be combined if it's a letter. Since you press ESC, it will just emit the circumflex on its own so Godot will see two inputs: the circumflex and the ESC.

This might be a platform-specific issue though. Can someone test on Linux and macOS?

**Comment by robfram** [comment]
Using X11 in FreeBSD with a Spanish key layout I get no keystroke registered with dead key and special keys like ESC or TAB, but another keycode with SPACE (which I assume is the dead key character itself).
If the dead key can't be combined with the pressed key after it (i.e. no output in X11), then no key is registered as being press.

For example:
`'` + `a` = `á` -> got one keystroke
`'`+ `t` = no output -> no keystroke

**Comment by Calinou** [comment]
@PetePete1984 Can you (or anyone else) still reproduce this bug in Godot 3.2.1 or [3.2.2beta4](https://godotengine.org/article/dev-snapshot-godot-3-2-2-beta-4)?

**Comment by PetePete1984** [comment]
Yeah, it still happens in 3.2.2 - you get the expected pressed / released events for the dead key, then 3 events for ESC (2x pressed, 1x released).

**Comment by CherrySodaPop** [comment]
I can confirm this is still an issue on 4.0 beta 3 Linux, frighteningly it also seems to happen with cursor movement somehow.

**Comment by growingbrain** [comment]
Still an issue in Godot 4 (v4.1.3.stable.official [f06b6836a]).
I have a local multiplayer game that has up to 4 controllers and keyboard input.

When the Escape key on the keyboard is pressed, I get up to 5 escape input just pressed signals; one after another.

If just keyboard is inserted (no extra controllers) then I only get 1 escape key signal.

So for some reason the escape key signal is sent for each device.
